test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java

```diff
@@ -0,0 +1,81 @@
+import jdk.test.lib.process.OutputAnalyzer;
+
+import java.io.*;
+import java.util.regex.Pattern;
+
+public class HsErrFileUtils {
+
+    /**
+     * Given the output of a java VM that crashed, extract the name of the hs-err file from the output
+     */
+    static public String extractHsErrFileNameFromOutput(OutputAnalyzer output) {
+        output.shouldMatch("# A fatal error has been detected.*");
+
+        // extract hs-err file
+        String hs_err_file = output.firstMatch("# *(\\S*hs_err_pid\\d+\\.log)", 1);
```

Comment:
I'm not sure this is going to be useful in the way you are trying to use it. This will show the original path to the hs_err file at the time it is created. But jtreg can move things around in the final test result output and place the hs_err file somewhere else.

Reply:
We use this pattern in a number of places, e.g. in BadNativeStackInErrorHandlingTest and SafeFetchInErrorHandlingTest.java. Seems to work ok so far.

Reply:
hmm - okay. I'm surprised.